cheatsheets/Content_Security_Policy_Cheat_Sheet.md

```diff
@@ -85,24 +91,37 @@ Navigation directives instruct the browser about the locations that the document
+- `report-uri` directive is deprecated by `report-to`, which is a URI that the reports are sent to.
   - Goes by the format of: `Content-Security-Policy: report-uri https://example.com/csp-reports`
 
 In order to ensure backward compatibility, use the 2 directives in conjonction. Whenever a browser supports `report-to`, it will ignore `report-uri`. Otherwise, `report-uri` will be used.
 
 ## Special Directive Sources
 
-| Value           | Description                                                     |
-|-----------------|-----------------------------------------------------------------|
-| 'none'          | No URLs match.                                                  |
-| 'self'          | Refers to the origin site with the same scheme and port number. |
-| 'unsafe-inline' | Allows the usage of inline scripts or styles.                   |
-| 'unsafe-eval'   | Allows the usage of eval in scripts.                            |
+| Value            | Description                                                                                                           |
+|------------------|-----------------------------------------------------------------------------------------------------------------------|
+| 'none'           | No URLs match.                                                                                                        |
+| 'self'           | Refers to the origin site with the same scheme and port number.                                                       |
+| 'unsafe-inline'  | Allows the usage of inline scripts or styles.                                                                         |
+| 'unsafe-eval'    | Allows the usage of eval in scripts.                                                                                  |
+| 'strict-dynamic' | Combining it with hashes or nonces, this tells the browser to trust scripts originating from the root trusted script. |
+
+In case where the developer needs to use inline scripts, it's recommended to use `hashes` for static scripts or a `nonce` on every page request.
 
-In case where the developer needs to use inline scripts, it's recommended to use `sha256` for the script or a `nonce` randomly generated on every page request. 
+To create hashes, check out this [hash generator](https://report-uri.com/home/hash). This is a great [example](https://csp.withgoogle.com/docs/faq.html#static-content) of using hashes.
 
-For more details on hashes and nonces, check out [Scott Helme's Guide](https://scotthelme.co.uk/csp-cheat-sheet/#hashes).
+To better understand how the directive sources work, check out the [source lists from w3c](https://w3c.github.io/webappsec-csp/#framework-directive-source-list).
+
+## Nonces
+
+[Nonces](https://en.wikipedia.org/wiki/Cryptographic_nonce) attributes are added to script tags. Nonce attributes are composed of base64 values. This nonce is verified against the nonce sent in the CSP header, and only matching nonces are allowed to execute.
```

Comment:
I have not found in the [CSP2](https://www.w3.org/TR/CSP2/) / [CSP3](https://www.w3.org/TR/CSP3/) doc a note about the fact that the nonce have a base64 value but perhaps i have missed it. 

From [CSP2](https://www.w3.org/TR/CSP2/#script-src-the-nonce-attribute) and [MDN](https://developer.mozilla.org/en-US/docs/Web/API/DOMString) docs:

```js
partial interface HTMLScriptElement {
  attribute DOMString nonce;
};
```

You talk about the base64 character sets because `nonce` seems be a random valid UTF-16 string?

Reply:
https://w3c.github.io/webappsec-csp/#framework-directive-source-list
It is in the format of 'nonce-'<base64 value>

Reply:
OK thank you for the feedback, indeed it's:

```
; Nonces: 'nonce-[nonce goes here]'
nonce-source  = "'nonce-" base64-value "'"
base64-value  = 1*( ALPHA / DIGIT / "+" / "/" / "-" / "_" )*2( "=" )
```